viewer:
  eye: !!python/tuple
  - 7.5
  - 7.5
  - 7.5
  lookat: !!python/tuple
  - 0.0
  - 0.0
  - 0.0
  cam_prim_path: /OmniverseKit_Persp
  resolution: !!python/tuple
  - 1280
  - 720
  origin_type: world
  env_index: 0
  asset_name: null
  body_name: null
sim:
  physics_prim_path: /physicsScene
  device: cuda:0
  dt: 0.008333333333333333
  render_interval: 2
  gravity: !!python/tuple
  - 0.0
  - 0.0
  - -9.81
  enable_scene_query_support: false
  use_fabric: true
  physx:
    solver_type: 1
    min_position_iteration_count: 1
    max_position_iteration_count: 255
    min_velocity_iteration_count: 0
    max_velocity_iteration_count: 255
    enable_ccd: false
    enable_stabilization: false
    enable_enhanced_determinism: false
    bounce_threshold_velocity: 0.5
    friction_offset_threshold: 0.04
    friction_correlation_distance: 0.025
    gpu_max_rigid_contact_count: 8388608
    gpu_max_rigid_patch_count: 163840
    gpu_found_lost_pairs_capacity: 2097152
    gpu_found_lost_aggregate_pairs_capacity: 33554432
    gpu_total_aggregate_pairs_capacity: 2097152
    gpu_collision_stack_size: 67108864
    gpu_heap_capacity: 67108864
    gpu_temp_buffer_capacity: 16777216
    gpu_max_num_partitions: 8
    gpu_max_soft_body_contacts: 1048576
    gpu_max_particle_contacts: 1048576
    solve_articulation_contact_last: false
  physics_material:
    func: isaaclab.sim.spawners.materials.physics_materials:spawn_rigid_body_material
    static_friction: 0.5
    dynamic_friction: 0.5
    restitution: 0.0
    friction_combine_mode: average
    restitution_combine_mode: average
    compliant_contact_stiffness: 0.0
    compliant_contact_damping: 0.0
  render:
    enable_translucency: null
    enable_reflections: null
    enable_global_illumination: null
    antialiasing_mode: null
    enable_dlssg: null
    enable_dl_denoiser: null
    dlss_mode: null
    enable_direct_lighting: null
    samples_per_pixel: null
    enable_shadows: null
    enable_ambient_occlusion: null
    dome_light_upper_lower_strategy: null
    carb_settings: null
    rendering_mode: null
  create_stage_in_memory: false
ui_window_class_type: isaaclab.envs.ui.base_env_window:BaseEnvWindow
seed: 42
decimation: 2
is_finite_horizon: false
episode_length_s: 12.0
scene:
  num_envs: 4096
  env_spacing: 4.0
  lazy_sensor_update: true
  replicate_physics: true
  filter_collisions: true
  clone_in_fabric: false
events: null
observation_space: 6
num_observations: null
state_space: 0
num_states: null
observation_noise_model: null
action_space: 1
num_actions: null
action_noise_model: null
rerender_on_reset: false
wait_for_textures: true
xr: null
log_dir: /home/jliu/isaaclab_ws/logs/rsl_rl/cartpole_direct/2025-12-02_14-35-42_foundation_layer4
robot_cfg:
  class_type: isaaclab.assets.articulation.articulation:Articulation
  prim_path: /World/envs/env_.*/Robot
  spawn:
    func: isaaclab.sim.spawners.from_files.from_files:spawn_from_usd
    visible: true
    semantic_tags: null
    copy_from_source: true
    mass_props: null
    deformable_props: null
    rigid_props:
      rigid_body_enabled: true
      kinematic_enabled: null
      disable_gravity: null
      linear_damping: null
      angular_damping: null
      max_linear_velocity: 1000.0
      max_angular_velocity: 1000.0
      max_depenetration_velocity: 100.0
      max_contact_impulse: null
      enable_gyroscopic_forces: true
      retain_accelerations: null
      solver_position_iteration_count: null
      solver_velocity_iteration_count: null
      sleep_threshold: null
      stabilization_threshold: null
    collision_props: null
    activate_contact_sensors: false
    scale: null
    articulation_props:
      articulation_enabled: null
      enabled_self_collisions: false
      solver_position_iteration_count: 4
      solver_velocity_iteration_count: 0
      sleep_threshold: 0.005
      stabilization_threshold: 0.001
      fix_root_link: null
    fixed_tendons_props: null
    spatial_tendons_props: null
    joint_drive_props: null
    visual_material_path: material
    visual_material: null
    usd_path: https://omniverse-content-production.s3-us-west-2.amazonaws.com/Assets/Isaac/5.1/Isaac/IsaacLab/Robots/Classic/CartDoublePendulum/cart_double_pendulum.usd
    variants: null
  init_state:
    pos: !!python/tuple
    - 0.0
    - 0.0
    - 2.0
    rot: !!python/tuple
    - 1.0
    - 0.0
    - 0.0
    - 0.0
    lin_vel: !!python/tuple
    - 0.0
    - 0.0
    - 0.0
    ang_vel: !!python/tuple
    - 0.0
    - 0.0
    - 0.0
    joint_pos:
      slider_to_cart: 0.0
      cart_to_pole: 0.0
      pole_to_pendulum: 0.0
    joint_vel:
      .*: 0.0
  collision_group: 0
  debug_vis: false
  articulation_root_prim_path: null
  soft_joint_pos_limit_factor: 1.0
  actuators:
    cart_actuator:
      class_type: isaaclab.actuators.actuator_pd:ImplicitActuator
      joint_names_expr:
      - slider_to_cart
      effort_limit: null
      velocity_limit: null
      effort_limit_sim: 400.0
      velocity_limit_sim: null
      stiffness: 0.0
      damping: 10.0
      armature: null
      friction: null
      dynamic_friction: null
      viscous_friction: null
    pole_actuator:
      class_type: isaaclab.actuators.actuator_pd:ImplicitActuator
      joint_names_expr:
      - cart_to_pole
      effort_limit: null
      velocity_limit: null
      effort_limit_sim: 400.0
      velocity_limit_sim: null
      stiffness: 0.0
      damping: 0.0
      armature: null
      friction: null
      dynamic_friction: null
      viscous_friction: null
    pendulum_actuator:
      class_type: isaaclab.actuators.actuator_pd:ImplicitActuator
      joint_names_expr:
      - pole_to_pendulum
      effort_limit: null
      velocity_limit: null
      effort_limit_sim: 400.0
      velocity_limit_sim: null
      stiffness: 0.0
      damping: 0.0
      armature: null
      friction: null
      dynamic_friction: null
      viscous_friction: null
  actuator_value_resolution_debug_print: false
cart_dof_name: slider_to_cart
pole1_dof_name: cart_to_pole
pole2_dof_name: pole_to_pendulum
action_scale: 25.0
target_pole1_angle: 0
target_pole2_angle: 0
target_cart_pos: 0.0
max_pole_angle_error: 3.141592653589793
max_cart_pos: 4
pole1_vel_weight: 1.5
pole2_vel_weight: 1.5
cart_vel_weight: 1.5
pole1_pos_weight: 1.5
pole2_pos_weight: 1.5
cart_pos_weight: 0.3
rew_scale_alive: 1.0
rew_scale_terminated: -3.0
rew_scale_joint_pos: -1.0
rew_scale_joint_vel: -0.01
initial_joint_angle_range:
- -2
- 2
